### PR TITLE
BaSP-TG-81: Update Project Forms

### PR DESCRIPTION
**Pull request by AlejoGF**
Implement React Hook Form on Projects Add, AddMember and Edit Forms.
Evaluate the operation, not the design since this will receive modifications in the modals

### CODE REVIEW

File: src/Components/Admin/Projects/Form/AddMember/AddMember.jsx [hunk 1]
```diff
@@ -106,45 +99,53 @@ const AddMember = ({ itemId, functionValue }) => {
...
+    formState: { errors, isDirty }
+  } = useForm({
+    mode: 'onChange',
+    resolver: joiResolver(schema),
+    defaultValues: {
+      member: '',
+      role: ''
+    }
+  });
+  console.log('error', errors);
```
Comment (LuliBianchini) [inline src/Components/Admin/Projects/Form/AddMember/AddMember.jsx:114]: There's a console.log left in this line

File: src/Components/Admin/Projects/Form/index.js [hunk 2]
```diff
@@ -4,70 +4,75 @@ import { getProjectById, updateProject } from 'redux/projects/thunks';
...
+  const handleOnSubmit = (data) => {
+    if (
+      data.name === project.name &&
+      data.startDate.toString() == new Date(project.startDate) &&
+      data.endDate.toString() == new Date(project.endDate) &&
+      data.clientName === project.clientName &&
+      data.active === project.active &&
+      data.description === project.description
+    ) {
+      setAlertMessage({ message: "There haven't been string changes", data: {}, error: true });
```
Comment (LuliBianchini) [inline src/Components/Admin/Projects/Form/index.js:67]: Maybe the message could be "There hasn't been any changes"?

File: src/Components/Admin/Tasks/Edit/Edit.jsx [hunk 3]
```diff
@@ -82,6 +86,7 @@ const Edit = ({ task, closeModalForm }) => {
         title="Choose status"
         data={['To do', 'In progress', 'Review', 'Blocked', 'Done', 'Cancelled']}
         required={true}
+        register={console.log}
       />
       <ButtonText clickAction={closeModalForm} label="Cancel">
         Cancel
```
Comment (LuliBianchini) [inline src/Components/Admin/Tasks/Edit/Edit.jsx:92]: This line can be erased since the button is labeled

**Review by LuliBianchini — COMMENTED** [review]
Everything is working and looking well! There are just a few things to delete

File: src/Components/Admin/Tasks/Edit/Edit.jsx (hunk 3 above)
Comment (LuliBianchini) [reply src/Components/Admin/Tasks/Edit/Edit.jsx:92]: Just in case, I meant the "Cancel" below the button

Comment (AlejoGF) [reply src/Components/Admin/Tasks/Edit/Edit.jsx:92]: I didn't work on Admin components.

File: src/Components/Admin/Projects/Form/AddMember/AddMember.jsx [hunk 4]
```diff
@@ -24,49 +37,36 @@ const AddMember = ({ itemId, functionValue }) => {
...
       }
     }
     setProjectMembers(projectMembers);
     return projectMembers;
   };
 
-  const handleOnSubmit = async () => {
-    if (member !== '' && role !== '' && rate !== '') {
+  const handleOnSubmit = (data) => {
+    console.log('data', data);
```
Comment (Johsua30) [inline src/Components/Admin/Projects/Form/AddMember/AddMember.jsx:66]: There's a console.log left in this line

**Review by Johsua30 — COMMENTED** [review]
A few messages should be changed but everything works fine

**Review by LuliBianchini — APPROVED** [review]
Everything looking good!

**Review by NicolasFontana — APPROVED** [review]
Everything looks good! Good work

**Comment by tinchorova** [comment]
## Test results

Tested locally on: https://alexis-trackgenix-app-git-feature-tg-81-updat-7112f0-basp-m2022.vercel.app/admin/projects

**Test result: Failed** ⚠️ 

Bugs found:

**Add project**
1. UI bugs: Inputs should be equidistant to the sides. Create button outside of the form. The modal creates a scrollbar.
image
2. There should be an 'on blur' event listener.
3. The invalid message shouldn't appear instantly when the user inputs a wrong data, (this is connected to point .2)
image
image
4. There should be mores space between labels and messages
image
5. If end date is determined already, and an invalid start date is selected it does not show an error message on start date. The logic of this validation may cause you trouble.
image
image
image

**Edit project**
1. Invalid messages are changing the size of the form and moving the inputs
image
image
2. Wording bug: 'rate' should have capital letter
image
3. There should be more space between labels and messages. Invalid messages are changing the size of the window(just a little bit)
image

**Review by tinchorova — CHANGES_REQUESTED** [review]
Please refer to: https://github.com/BaSP-m2022/alexis-trackgenix-app/pull/39#issuecomment-1166066330

**Comment by AlejoGF** [comment]
Please do the tests in 1440x1024

**Review by tinchorova — APPROVED** [review]

**Pull request merged**